src/run.ts

```diff
@@ -67,10 +67,20 @@ export async function run() {
         await deploy(new Kubectl(kubectlPath, namespace), manifests, strategy);
     }
     else if (action === 'promote') {
-        await promote(true);
+        let strategy = core.getInput('strategy');
+        if(strategy==='canary') {
+            await promote(true);

```

Comment:
I don't see you sending any other input.. if it's true by default; why would you want to have a parameter?

Reply:
Same applies for all the newly introduced functions

Reply:
promote already exists right? even i was confused about the parameter. ill change and remove it